src/sgame/sg_combat.cpp

```diff
@@ -158,19 +158,25 @@ static const gentity_t *G_FindKillAssist( const gentity_t *self, const gentity_t
 	int             playerNum;
 
 	// Suicide? No assistance needed with that
-	if ( killer == self)
+	if ( killer == self )
+	{
+		return nullptr;
+	}
+	// Killed by an entity that is not a client? Ignore that too
+	if ( killer - g_entities >= MAX_CLIENTS )
 	{
 		return nullptr;
 	}
 
-	// Require that the assist was for, at least, 25% of the damage or
+	// Require that the assist was for, at least, 25% of the health or
 	// as much damage as the killer did, whichever is lower
 	damage = self->entity->Get<HealthComponent>()->MaxHealth() / 4.0f;
 	if ( killer )
```

Comment:
I don't see why there shouldn't be assists when the killer is a non-client (or the same player who died for that matter...). Why not just change `if ( killer )` to `if ( killer && killer->client )`

Reply:
Fair enough, it makes sense to have kill assist messages when something is killed by, say, a turret. Fixed.